File: 30_days_C#_HackerRank_Challenge/Sorting.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution
{
    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> a = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(aTemp => Convert.ToInt32(aTemp)).ToList();

        int totalSwaps = 0;

        for (int i = 0; i < n; i++)
        {
            int numberOfSwaps = 0;

            for (int j = 0; j < n - 1; j++)
            {
                if (a[j] > a[j + 1])
                {
                    int temp = a[j];
                    a[j] = a[j + 1];
                    a[j + 1] = temp;

                    numberOfSwaps++;
                    totalSwaps++;
                }
            }

            if (numberOfSwaps == 0)
                break;
        }

        Console.WriteLine($"Array is sorted in {totalSwaps} swaps.");
        Console.WriteLine($"First Element: {a[0]}");
        Console.WriteLine($"Last Element: {a[n - 1]}");
    }
}

File: 30_days_Python_HackerRank_Challenge/Sorting.py
#python-2
#!/bin/python

import math
import os
import random
import re
import sys

if __name__ == '__main__':
    n = int(raw_input().strip())

    a = map(int, raw_input().rstrip().split())

totalSwaps = 0

for i in range(n):
    numberOfSwaps = 0

    for j in range(n - 1):
        if a[j] > a[j + 1]:
            a[j], a[j + 1] = a[j + 1], a[j]
            numberOfSwaps += 1
            totalSwaps += 1

    if numberOfSwaps == 0:
        break

print "Array is sorted in %d swaps." % totalSwaps
print "First Element: %d" % a[0]
print "Last Element: %d" % a[n - 1]


#python-3
#!/bin/python3

import math
import os
import random
import re
import sys

if __name__ == '__main__':
    n = int(input().strip())

    a = list(map(int, input().rstrip().split()))

    totalSwaps = 0

for i in range(n):
    numberOfSwaps = 0

    for j in range(n - 1):
        if a[j] > a[j + 1]:
            a[j], a[j + 1] = a[j + 1], a[j]
            numberOfSwaps += 1
            totalSwaps += 1

    if numberOfSwaps == 0:
        break

print(f"Array is sorted in {totalSwaps} swaps.")
print(f"First Element: {a[0]}")
print(f"Last Element: {a[-1]}")
